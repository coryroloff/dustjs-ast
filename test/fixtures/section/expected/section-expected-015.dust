{#section myparam="starting {~lb} right here {one.two} {~rb} ending over here"}{/section}