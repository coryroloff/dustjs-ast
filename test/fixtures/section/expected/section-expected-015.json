{
	"type": "Template",
	"body": [
		{
			"type": "Section",
			"body": [
				{
					"type": "Block",
					"body": []
				}
			],
			"params": [
				{
					"type": "Param",
					"key": {
						"type": "Key",
						"name": "myparam",
						"start": 10,
						"end": 17,
						"loc": {
							"start": {
								"line": 1,
								"column": 10
							},
							"end": {
								"line": 1,
								"column": 17
							}
						}
					},
					"value": {
						"type": "Inline",
						"body": [
							{
								"type": "StringLiteral",
								"value": "starting ",
								"start": 19,
								"end": 28,
								"loc": {
									"start": {
										"line": 2,
										"column": 19
									},
									"end": {
										"line": 2,
										"column": 28
									}
								}
							},
							{
								"type": "Special",
								"key": {
									"type": "Key",
									"name": "lb",
									"start": 30,
									"end": 32,
									"loc": {
										"start": {
											"line": 2,
											"column": 30
										},
										"end": {
											"line": 2,
											"column": 32
										}
									}
								},
								"start": 28,
								"end": 33,
								"loc": {
									"start": {
										"line": 2,
										"column": 28
									},
									"end": {
										"line": 2,
										"column": 33
									}
								}
							},
							{
								"type": "StringLiteral",
								"value": " right here ",
								"start": 33,
								"end": 45,
								"loc": {
									"start": {
										"line": 2,
										"column": 33
									},
									"end": {
										"line": 2,
										"column": 45
									}
								}
							},
							{
								"type": "Reference",
								"id": {
									"type": "Identifier",
									"path": [
										{
											"type": "Key",
											"name": "one",
											"start": 46,
											"end": 49,
											"loc": {
												"start": {
													"line": 2,
													"column": 46
												},
												"end": {
													"line": 2,
													"column": 49
												}
											}
										},
										{
											"type": "Key",
											"name": "two",
											"start": 50,
											"end": 53,
											"loc": {
												"start": {
													"line": 2,
													"column": 50
												},
												"end": {
													"line": 2,
													"column": 53
												}
											}
										}
									],
									"localContext": false,
									"start": 46,
									"end": 53,
									"loc": {
										"start": {
											"line": 2,
											"column": 46
										},
										"end": {
											"line": 2,
											"column": 53
										}
									}
								},
								"filters": [],
								"start": 45,
								"end": 54,
								"loc": {
									"start": {
										"line": 2,
										"column": 45
									},
									"end": {
										"line": 2,
										"column": 54
									}
								}
							},
							{
								"type": "StringLiteral",
								"value": " ",
								"start": 54,
								"end": 55,
								"loc": {
									"start": {
										"line": 2,
										"column": 54
									},
									"end": {
										"line": 2,
										"column": 55
									}
								}
							},
							{
								"type": "Special",
								"key": {
									"type": "Key",
									"name": "rb",
									"start": 57,
									"end": 59,
									"loc": {
										"start": {
											"line": 2,
											"column": 57
										},
										"end": {
											"line": 2,
											"column": 59
										}
									}
								},
								"start": 55,
								"end": 60,
								"loc": {
									"start": {
										"line": 2,
										"column": 55
									},
									"end": {
										"line": 2,
										"column": 60
									}
								}
							},
							{
								"type": "StringLiteral",
								"value": " ending over here",
								"start": 60,
								"end": 77,
								"loc": {
									"start": {
										"line": 2,
										"column": 60
									},
									"end": {
										"line": 1,
										"column": 77
									}
								}
							}
						],
						"start": 18,
						"end": 78,
						"loc": {
							"start": {
								"line": 1,
								"column": 18
							},
							"end": {
								"line": 1,
								"column": 78
							}
						}
					},
					"start": 10,
					"end": 78,
					"loc": {
						"start": {
							"line": 1,
							"column": 10
						},
						"end": {
							"line": 1,
							"column": 78
						}
					}
				}
			],
			"format": {
				"startTagAfterStart": null,
				"startTagBeforeParam": [
					{
						"type": "Format",
						"value": " ",
						"start": 9,
						"end": 10,
						"loc": {
							"start": {
								"line": 1,
								"column": 9
							},
							"end": {
								"line": 1,
								"column": 10
							}
						}
					}
				],
				"startTagBeforeEnd": null,
				"endTagAfterStart": null,
				"endTagBeforeEnd": null
			},
			"kind": "#",
			"id": {
				"type": "Identifier",
				"path": [
					{
						"type": "Key",
						"name": "section",
						"start": 2,
						"end": 9,
						"loc": {
							"start": {
								"line": 1,
								"column": 2
							},
							"end": {
								"line": 1,
								"column": 9
							}
						}
					}
				],
				"localContext": false,
				"start": 2,
				"end": 9,
				"loc": {
					"start": {
						"line": 1,
						"column": 2
					},
					"end": {
						"line": 1,
						"column": 9
					}
				}
			},
			"start": 0,
			"end": 89,
			"loc": {
				"start": {
					"line": 1,
					"column": 0
				},
				"end": {
					"line": 1,
					"column": 89
				}
			}
		}
	],
	"tokens": [
		{
			"type": "Punctuator",
			"value": "{",
			"start": 0,
			"end": 1,
			"loc": {
				"start": {
					"line": 1,
					"column": 0
				},
				"end": {
					"line": 1,
					"column": 1
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "#",
			"start": 1,
			"end": 2,
			"loc": {
				"start": {
					"line": 1,
					"column": 1
				},
				"end": {
					"line": 1,
					"column": 2
				}
			}
		},
		{
			"type": "Key",
			"value": "section",
			"start": 2,
			"end": 9,
			"loc": {
				"start": {
					"line": 1,
					"column": 2
				},
				"end": {
					"line": 1,
					"column": 9
				}
			}
		},
		{
			"type": "Format",
			"value": " ",
			"start": 9,
			"end": 10,
			"loc": {
				"start": {
					"line": 1,
					"column": 9
				},
				"end": {
					"line": 1,
					"column": 10
				}
			}
		},
		{
			"type": "Key",
			"value": "myparam",
			"start": 10,
			"end": 17,
			"loc": {
				"start": {
					"line": 1,
					"column": 10
				},
				"end": {
					"line": 1,
					"column": 17
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "=",
			"start": 17,
			"end": 18,
			"loc": {
				"start": {
					"line": 1,
					"column": 17
				},
				"end": {
					"line": 1,
					"column": 18
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "\"",
			"start": 18,
			"end": 19,
			"loc": {
				"start": {
					"line": 1,
					"column": 18
				},
				"end": {
					"line": 2,
					"column": 19
				}
			}
		},
		{
			"type": "StringLiteral",
			"value": "starting ",
			"start": 19,
			"end": 28,
			"loc": {
				"start": {
					"line": 2,
					"column": 19
				},
				"end": {
					"line": 2,
					"column": 28
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "{",
			"start": 28,
			"end": 29,
			"loc": {
				"start": {
					"line": 2,
					"column": 28
				},
				"end": {
					"line": 2,
					"column": 29
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "~",
			"start": 29,
			"end": 30,
			"loc": {
				"start": {
					"line": 2,
					"column": 29
				},
				"end": {
					"line": 2,
					"column": 30
				}
			}
		},
		{
			"type": "Key",
			"value": "lb",
			"start": 30,
			"end": 32,
			"loc": {
				"start": {
					"line": 2,
					"column": 30
				},
				"end": {
					"line": 2,
					"column": 32
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "}",
			"start": 32,
			"end": 33,
			"loc": {
				"start": {
					"line": 2,
					"column": 32
				},
				"end": {
					"line": 2,
					"column": 33
				}
			}
		},
		{
			"type": "StringLiteral",
			"value": " right here ",
			"start": 33,
			"end": 45,
			"loc": {
				"start": {
					"line": 2,
					"column": 33
				},
				"end": {
					"line": 2,
					"column": 45
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "{",
			"start": 45,
			"end": 46,
			"loc": {
				"start": {
					"line": 2,
					"column": 45
				},
				"end": {
					"line": 2,
					"column": 46
				}
			}
		},
		{
			"type": "Key",
			"value": "one",
			"start": 46,
			"end": 49,
			"loc": {
				"start": {
					"line": 2,
					"column": 46
				},
				"end": {
					"line": 2,
					"column": 49
				}
			}
		},
		{
			"type": "Punctuator",
			"value": ".",
			"start": 49,
			"end": 50,
			"loc": {
				"start": {
					"line": 2,
					"column": 49
				},
				"end": {
					"line": 2,
					"column": 50
				}
			}
		},
		{
			"type": "Key",
			"value": "two",
			"start": 50,
			"end": 53,
			"loc": {
				"start": {
					"line": 2,
					"column": 50
				},
				"end": {
					"line": 2,
					"column": 53
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "}",
			"start": 53,
			"end": 54,
			"loc": {
				"start": {
					"line": 2,
					"column": 53
				},
				"end": {
					"line": 2,
					"column": 54
				}
			}
		},
		{
			"type": "StringLiteral",
			"value": " ",
			"start": 54,
			"end": 55,
			"loc": {
				"start": {
					"line": 2,
					"column": 54
				},
				"end": {
					"line": 2,
					"column": 55
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "{",
			"start": 55,
			"end": 56,
			"loc": {
				"start": {
					"line": 2,
					"column": 55
				},
				"end": {
					"line": 2,
					"column": 56
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "~",
			"start": 56,
			"end": 57,
			"loc": {
				"start": {
					"line": 2,
					"column": 56
				},
				"end": {
					"line": 2,
					"column": 57
				}
			}
		},
		{
			"type": "Key",
			"value": "rb",
			"start": 57,
			"end": 59,
			"loc": {
				"start": {
					"line": 2,
					"column": 57
				},
				"end": {
					"line": 2,
					"column": 59
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "}",
			"start": 59,
			"end": 60,
			"loc": {
				"start": {
					"line": 2,
					"column": 59
				},
				"end": {
					"line": 2,
					"column": 60
				}
			}
		},
		{
			"type": "StringLiteral",
			"value": " ending over here",
			"start": 60,
			"end": 77,
			"loc": {
				"start": {
					"line": 2,
					"column": 60
				},
				"end": {
					"line": 1,
					"column": 77
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "\"",
			"start": 77,
			"end": 78,
			"loc": {
				"start": {
					"line": 1,
					"column": 77
				},
				"end": {
					"line": 1,
					"column": 78
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "}",
			"start": 78,
			"end": 79,
			"loc": {
				"start": {
					"line": 1,
					"column": 78
				},
				"end": {
					"line": 1,
					"column": 79
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "{",
			"start": 79,
			"end": 80,
			"loc": {
				"start": {
					"line": 1,
					"column": 79
				},
				"end": {
					"line": 1,
					"column": 80
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "/",
			"start": 80,
			"end": 81,
			"loc": {
				"start": {
					"line": 1,
					"column": 80
				},
				"end": {
					"line": 1,
					"column": 81
				}
			}
		},
		{
			"type": "Key",
			"value": "section",
			"start": 81,
			"end": 88,
			"loc": {
				"start": {
					"line": 1,
					"column": 81
				},
				"end": {
					"line": 1,
					"column": 88
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "}",
			"start": 88,
			"end": 89,
			"loc": {
				"start": {
					"line": 1,
					"column": 88
				},
				"end": {
					"line": 1,
					"column": 89
				}
			}
		}
	],
	"start": 0,
	"end": 89,
	"loc": {
		"start": {
			"line": 1,
			"column": 0
		},
		"end": {
			"line": 1,
			"column": 89
		}
	}
}